djangocms_versioning_filer/monkeypatch/admin/folderadmin.py

```diff
@@ -314,17 +314,33 @@ def filer_add_items_to_collection(self, request, files_qs, folders_qs):
     return add_items_to_collection(self, request, files_qs)
 
 
+def filer_add_items_to_unpublish_collection(self, request, files_qs, folders_qs):
+    from djangocms_moderation.admin_actions import add_item_to_unpublish_collection
+    for folder in folders_qs:
+        files_qs |= get_files_distinct_grouper_queryset().filter(
+            folder__in=folder.get_descendants(include_self=True),
+        )
+    files_qs = files_qs.filter(versions__state=PUBLISHED)
+    return add_item_to_unpublish_collection(self, request, files_qs)
+
+
 def get_actions(func):
     def inner(self, request):
         actions = func(self, request)
 
         if is_moderation_enabled():
             from djangocms_moderation.admin_actions import add_items_to_collection
```

Comment:
I didn't realise that we had moderation dependence in this package. Probably shouldn't have been the case.

Reply:
Well, you kind of have to have that dependence if you want to add it to a moderation collection.